### PR TITLE
Processed data in models

### PR DESCRIPTION
**Pull request by alphayowakarindi**
At this juncture of development I implemented the following:
Create custom methods

-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments
+  after_save :update_posts_counter
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - In reading from the ERD, shows that Post has many like. Please add it.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```
Comment (OLIPLICHE) [inline app/models/post.rb]: - **Suggestion**: Well done in using "[Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)" `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database. Now you need to make make the `update_post_counter` **method private**

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: 'User'
   belongs_to :post
+  after_save :update_comments_counter
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (OLIPLICHE) [inline app/models/comment.rb:10]: - **Suggestion**:
- Make `update_comments_counter` **private method** as I commented in Post.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
   belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_save :update_likes_counter
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (OLIPLICHE) [inline app/models/like.rb:11]: - **Suggestion**:
- Here too the same: `Private method`

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @alphayowakarindi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used  👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alphayowakarindi** [comment]
Hi @OLIPLICHE, thank you so much for taking the time to leave this excellent review. I've implemented the changes and I really appreciate your input to make my code better. Cheers!

File: app/models/post.rb (hunk 1 above)
Comment (Whoistolu) [reply app/models/post.rb]: - Well resolved 👍🏽

File: app/models/post.rb (hunk 2 above)
Comment (Whoistolu) [reply app/models/post.rb]: -  Well resolved as well 👍🏽

File: app/models/comment.rb (hunk 3 above)
Comment (Whoistolu) [reply app/models/comment.rb:10]: - Well resolved 👍🏽

File: app/models/like.rb (hunk 4 above)
Comment (Whoistolu) [reply app/models/like.rb:11]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @alphayowakarindi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**